File: telo/src/main/java/com/hartwig/hmftools/telo/analysers/TelomereReadsAnalyser.java
package com.hartwig.hmftools.telo.analysers;

import java.util.HashMap;
import java.util.Map;

import com.hartwig.hmftools.telo.ReadGroup;
import com.hartwig.hmftools.telo.TeloUtils;

// analyse the telo bam and come up with our own score
// of F1, F2, F4
public class TelomereReadsAnalyser
{
    private final Map<ReadGroup.FragmentType, Integer> mFragmentTypeCount = new HashMap<>();

    public int getFragmentTypeCount(ReadGroup.FragmentType fragType)
    {
        return mFragmentTypeCount.getOrDefault(fragType, 0);
    }

    public void onReadGroup(final ReadGroup readGroup)
    {
        // we want to classify this read group
        ReadGroup.FragmentType fragType = TeloUtils.classifyFragment(readGroup);
        mFragmentTypeCount.put(fragType, mFragmentTypeCount.getOrDefault(fragType, 0) + 1);
    }
}

File: teal/src/main/java/com/hartwig/hmftools/teal/analysers/TelomereReadsAnalyser.kt
package com.hartwig.hmftools.teal.analysers

import com.hartwig.hmftools.teal.ReadGroup
import com.hartwig.hmftools.teal.TeloUtils
import java.util.*

// analyse the telo bam and come up with our own score
// of F1, F2, F4
class TelomereReadsAnalyser
{
    private val mFragmentTypeCount: MutableMap<ReadGroup.FragmentType, Int> = EnumMap(ReadGroup.FragmentType::class.java)
    fun getFragmentTypeCount(fragType: ReadGroup.FragmentType): Int
    {
        return mFragmentTypeCount.getOrDefault(fragType, 0)
    }

    fun onReadGroup(readGroup: ReadGroup?)
    {
        // we want to classify this read group
        val fragType = TeloUtils.classifyFragment(readGroup)
        mFragmentTypeCount[fragType] = mFragmentTypeCount.getOrDefault(fragType, 0) + 1
    }
}